packages/flutter_tools/templates/create/android.tmpl/buildSrc/src/main/groovy/FlutterPlugin.groovy

```diff
@@ -0,0 +1,229 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
```

Comment:
This is a lot of boilerplate. Is there any way we can factor this out into code that's reused?

Having a lot of code here means:
* If we ever have to change it, we'll lose any users who have already created an app with the old template.
* Users will have all this code they don't understand and didn't write in their codebase, and will likely fiddle with it in ways that break them and we won't understand what's wrong.

If we keep this code we should document the heck out of it, like we do the default main.dart template.